### DISCUSSION TITLE
Support custom uuid classes for Uuid type

### DISCUSSION THREAD

**Discussion by kasium** [sqlalchemy/sqlalchemy#11683] [Ideas]
For python, normally the stdlib is used to generate uuids. However, there are cases where a faster generation is needed. For this, libraries like [fastuuid](https://pypi.org/project/fastuuid/) exists.

It would be great, if the `Uuid` type would support the usage of other uuid classes if `as_uuid=True` is set.
Interface wise I could image a similar way as for `json_serializer/json_deserializer`, so it's set on the dialect. Else the `Uuid` type could receive a `uuid_class` parameter which specifies the uuid class to use

**Comment by CaselIT** [comment]
Hi,

While I'm not against it, typing it correctly is not that simple.

Overall though I think a type decorator that does what you want is likely better, since many driver return python uuid already, so correctly handling all cases in a generic way may bring q lot of complexity

**Reply by kasium** [reply]
I agree, typing will be a challenge. But hopefully the class will be compatible to uuid.UUID so just keeping that as the type is the right for the start

**Reply by CaselIT** [reply]
man complexity is probably supporting the additional cases that would result with the processors.

For example we could have a driver that returns a python uuid that we would need to convert to the provided uuid. same when inserting values.
Since performance here is mentioned users would likely want to configure the driver to directly create the 3-rd party class, so now the conversion done by the class would not be needed.
I'm not sure if supporting all this in a general way is worth it

@zzzeek thoughts on this?
